web/models/user.ex

```diff
@@ -9,22 +12,56 @@ defmodule Yggdrasil.User do
     timestamps
   end
 
-  @required_fields ~w(username password password_confirmation)
-  @optional_fields ~w()
-
   @doc """
-  Creates a changeset based on the `model` and `params`.
-
-  If no params are provided, an invalid changeset is returned
-  with no validation performed.
+  create_changeset should be used when creating a new user it ensures
+  all the fields needed are there and valid and generates a password
   """
-  def changeset(model, params \\ :empty) do
+  def create_changeset(model, params \\ :empty) do
     model
-    |> cast(params, @required_fields, @optional_fields)
+    |> cast(params, ~w(username password password_confirmation), ~w())
     |> update_change(:username, &String.downcase/1)
     |> unique_constraint(:username)
     |> validate_length(:password, min: 4)
     |> validate_length(:password_confirmation, min: 4)
     |> validate_confirmation(:password)
+    |> add_password
+  end
+
+  @doc """
+  update_pw_changeset shuld be used when updating a password for a user.
+  it will only change the password field and ensures that username and other
+  fields are not modified.
+  """
+  def update_pw_changeset(model, params \\ :empty) do
```

Comment:
If these are not used by the app I would remove them. That way we don't get caught up writing tests or maintaining these until they are needed.